DynamicTablesPkg/Library/Acpi/Common/AcpiFadtLib/Common/CommonFadtGenerator.c

```diff
@@ -41,6 +125,262 @@ FadtArchUpdate (
+  Status = GetEArchCommonObjFadtMiscInfo (
+             CfgMgrProtocol,
+             CM_NULL_TOKEN,
+             &FadtMiscInfo,
+             NULL
+             );
+  if (EFI_ERROR (Status)) {
+    DEBUG ((
+      DEBUG_ERROR,
+      "ERROR: FADT: Failed to get various platform information." \
+      " Status = %r\n",
+      Status
+      ));
+  } else {
+    Fadt->PLvl2Lat    = FadtMiscInfo->PLvl2Lat;
+    Fadt->PLvl3Lat    = FadtMiscInfo->PLvl3Lat;
+    Fadt->FlushSize   = FadtMiscInfo->FlushSize;
+    Fadt->FlushStride = FadtMiscInfo->FlushStride;
+    Fadt->DutyOffset  = FadtMiscInfo->DutyOffset;
+    Fadt->DutyWidth   = FadtMiscInfo->DutyWidth;
+    Fadt->DayAlrm     = FadtMiscInfo->DayAlrm;
+    Fadt->MonAlrm     = FadtMiscInfo->MonAlrm;
+    Fadt->Century     = FadtMiscInfo->Century;
+  }
+
+  // Get the 64-bit GPE Block information from the Platform Configuration Manager
+  Status = GetEArchCommonObjFadtXgpeBlockInfo (
+             CfgMgrProtocol,
+             CM_NULL_TOKEN,
+             &XgpeBlockInfo,
+             NULL
+             );
+  if (EFI_ERROR (Status)) {
+    DEBUG ((
+      DEBUG_ERROR,
+      "ERROR: FADT: Failed to get 64-bit GPE Block information." \
+      " Status = %r\n",
+      Status
+      ));
+  } else {
+    CopyMem (
+      &Fadt->XGpe0Blk,
+      &XgpeBlockInfo->XGpe0Blk,
+      sizeof (EFI_ACPI_6_5_GENERIC_ADDRESS_STRUCTURE)
+      );
+    CopyMem (
+      &Fadt->XGpe1Blk,
+      &XgpeBlockInfo->XGpe1Blk,
+      sizeof (EFI_ACPI_6_5_GENERIC_ADDRESS_STRUCTURE)
+      );
+  }
+
+  // Get the sleep Block information from the Platform Configuration Manager
+  Status = GetEArchCommonObjFadtSleepBlockInfo (
+             CfgMgrProtocol,
+             CM_NULL_TOKEN,
+             &SleepBlockInfo,
+             NULL
+             );
+  if (EFI_ERROR (Status)) {
+    DEBUG ((
+      DEBUG_ERROR,
+      "ERROR: FADT: Failed to get Sleep Block information." \
+      " Status = %r\n",
+      Status
+      ));
+  } else {
+    CopyMem (
+      &Fadt->SleepControlReg,
+      &SleepBlockInfo->SleepControlReg,
+      sizeof (EFI_ACPI_6_5_GENERIC_ADDRESS_STRUCTURE)
+      );
+    CopyMem (
+      &Fadt->SleepStatusReg,
+      &SleepBlockInfo->SleepStatusReg,
+      sizeof (EFI_ACPI_6_5_GENERIC_ADDRESS_STRUCTURE)
+      );
+  }
+
+  // Get the sleep Block information from the Platform Configuration Manager
+  Status = GetEArchCommonObjFadtResetBlockInfo (
+             CfgMgrProtocol,
+             CM_NULL_TOKEN,
+             &ResetBlockInfo,
+             NULL
+             );
+  if (EFI_ERROR (Status)) {
+    DEBUG ((
+      DEBUG_ERROR,
+      "ERROR: FADT: Failed to get Reset Block information." \
+      " Status = %r\n",
+      Status
+      ));
+  } else {
+    CopyMem (
+      &Fadt->ResetReg,
+      &ResetBlockInfo->ResetReg,
+      sizeof (EFI_ACPI_6_5_GENERIC_ADDRESS_STRUCTURE)
+      );
+    Fadt->ResetValue = ResetBlockInfo->ResetValue;
+  }
+
+  // Update Creator ID

```

Comment:
Hello Abdul,
I still need to discuss with Sami when he comes back from holiday about this and about this patch:
  `DynamicTablesPkg: Defines generic table generator creator id`
but from my point of view the patches looks good to me!